src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -255,16 +255,23 @@ namespace OpenMS
     // FEATURE : normalize with the intensity at the peak group apex?
     if (su_.use_shape_score_)
     {
-      scores.xcorr_shape_score = mrmscore_.calcXcorrShape_score();
-      scores.weighted_xcorr_shape = mrmscore_.calcXcorrShape_score_weighted(normalized_library_intensity);
+      scores.xcorr_shape_score = mrmscore_.calcXcorrShapeScore();
+      scores.weighted_xcorr_shape = mrmscore_.calcXcorrShapeWeightedScore(normalized_library_intensity);
     }
 
     // check that the MS1 feature is present and that the MS1 correlation should be calculated
     if (imrmfeature->getPrecursorIDs().size() > 0 && su_.use_ms1_correlation)
     {
-      mrmscore_.initializeMS1XCorr(imrmfeature, native_ids, precursor_feature_id); // perform cross-correlation on monoisotopic precursor
-      scores.xcorr_ms1_coelution_score = mrmscore_.calcMS1XcorrCoelutionScore();
-      scores.xcorr_ms1_shape_score = mrmscore_.calcMS1XcorrShape_score();
+      // we need at least two precursor isotopes
+      if (precursor_ids.size() > 1)
+      {
+        mrmscore_.initializeXCorrPrecursorMatrix(imrmfeature, precursor_ids);
+        scores.ms1_xcorr_coelution_score = mrmscore_.calcXcorrPrecursorCoelutionScore();
+        scores.ms1_xcorr_shape_score = mrmscore_.calcXcorrPrecursorShapeScore();
```

Comment:
what are the default values here (I think 0) -- however 0 is also the best possible score you can get. So in case you dont have any 2nd isotope, you get the best possible score? That may mess up the scoring.

Reply:
Indeed, this is very dangerous. It will work when only MS1 traces with a signal are supplied, but not otherwise. I will look into improving the exception management here.

Reply:
I am more worried about the scoring, lets assume 2 precursors:

1. precursor has low but detectable MS1_isotope1 signal, therefore gets scored and gets a score of 0.6
2. precursor has no detectable MS1_isotope1 signal, therefore Line 267 to 271 is skipped and it gets a default score of 0 -- which makes its score better than the other signal. I think that will lead to inflation of scores around zero

Reply:
I now implemented a failsafe to address this problem, however we might want to consider whether we should make this even more generic, see below.